play-json/shared/src/test/scala/JsonExtensionSpec.scala

```diff
@@ -712,5 +714,20 @@ class JsonExtensionSpec extends WordSpec with MustMatchers {
       "by reads macro" in validateReads(macroReads)
       "by format macro" in validateReads(macroFormat)
     }
+
+    "configuration methods" should {
+      val json = Json.obj("first_prop" -> "the first")
+      val data = WithDefaultSnake("the first")
+
+      "allow supplying configuration via implicit config" in {
+        implicit val config = JsonConfiguration[Json.WithDefaultValues](naming = SnakeCase)
+        json.as(Json.reads[WithDefaultSnake]) mustEqual data
+      }
+
+      "allow supplying configuration via WithOptions" in {
+        val config = JsonConfiguration[Json.WithDefaultValues](naming = SnakeCase)
+        json.as(Json.configured(config).reads[WithDefaultSnake]) mustEqual data
+      }
+    }
```

Comment:
And here's the results, we can see that both the naming, and the macro options, can be passed either via an implicit config, or via the `WithOptions` implementation of `JsonFacade`.